### DISCUSSION TITLE
Increase containers resources

### DISCUSSION THREAD

**Discussion by hosmelq** [basecamp/kamal#632] [Q&A]
I have a Laravel application running on a small server (1CPU, 1RAM) to test Kamal. When I made a stress test I got very slow responses, I checked the server resources and it does not consume 0.5 CPU even. Is there any way to let the containers consume all the available resources?

stress test output:

![CleanShot 2024-01-07 at 3  28 54@2x](https://github.com/basecamp/kamal/assets/1166143/367c83b0-5aa7-4627-849f-9638ea464422)

docker stats output during stress test:

![CleanShot 2024-01-07 at 3  30 33@2x](https://github.com/basecamp/kamal/assets/1166143/4f3016a8-ad31-498c-a813-970d810d9ee6)

**Comment by nickhammond** [comment]
@hosmelq Did you set cpu or memory limits on the container? Kamal doesn't restrict CPU or memory unless you set it as an option. If it's not set as an option it'll use as much as it needs.

How many web processes do you have set up to handle requests? It looks like your requests are taking 4s each so the problem looks to be application side and not on the Docker/Kamal side.
